pkg/router/template/router.go

```diff
@@ -1393,6 +1447,8 @@ func configsAreEqual(config1, config2 *ServiceAliasConfig) bool {
 		config1.RoutingKeyName == config2.RoutingKeyName &&
 		config1.IsWildcard == config2.IsWildcard &&
 		config1.VerifyServiceHostname == config2.VerifyServiceHostname &&
+		reflect.DeepEqual(config1.HTTPResponseHeaders, config2.HTTPResponseHeaders) &&
+		reflect.DeepEqual(config1.HTTPRequestHeaders, config2.HTTPRequestHeaders) &&
```

Comment:
This is something that should probably be unit tested.  Is it possible?

Reply:
I am not sure how we could test it. Just passing two equal and unequal  HTTPResponseHeaders and 
HTTPRequestHeaders  ServiceAliasConfig object ?

Reply:
I think there's going to be a problem here if the headers get out of order.  Some evidence that it works would be helpful.

Reply:
@candita, is the unit test in https://github.com/miheer/router/pull/2 satisfactory?